### DISCUSSION TITLE
Online meeting on Symbolic Regression April 28-29

### DISCUSSION THREAD

**Discussion by pukpr** [MilesCranmer/PySR#900] [General]
sounds interesting, free online but must register

see https://www.eventbrite.co.uk/e/symbolic-regression-in-the-physical-sciences-tickets-1110181545889

https://bsky.app/profile/royalsocietypublishing.org/post/3lnlcbri6bn2c

**Comment by pukpr** [comment]
Meeting is going on for a 2nd day. Here is an interesting exchange with an audience member on the first day,  who asked if the physics model they eventually used was found brute force by the symbolic regression:
https://youtu.be/fzVnkDSPwt0?t=9296

Found it strange that the speaker had not thought about this, as that seems like an obvious aspect to check and one of the strong rationales for doing symbolic regression.  

From my understanding, the speaker explained that they were using the symbolic regression to **emulate** the signal, perhaps as a noise reduction approach a la a spline curve, and that they were satisfied with that.

So far the videos are informative.

**Reply by pukpr** [reply]
Here is the 2nd day of the meeting
https://youtu.be/8o6jU-iBXbw
This is an unlisted YT video so not sure if it will stay around.

**Comment by gkronber** [comment]
Thanks for linking the youtube videos. The Royal Society will also provide videos of each talk individually after some minor editing. 

We prepared a summary from the meeting here: https://symreg.at/blog/2025/royal-society-discussion-meeting-symreg-in-physics/
Unfortunately, Miles couldn't stay until the end for the group picture. 

@MilesCranmer  thanks for your insightful talk. I especially like the flexibility of PySR, for instance that it allows any kind of custom loss function.

**Comment by pukpr** [comment]
Always looking for some secret sauce that will allow symbolic regression to find more complicated expressions that aren't overly penalized by a loss function. 

I gave this prompt to ChatGPT:
> In symbolic regression for discovering formula, to prevent  overfitting , a complexity measure is introduced which counts the number of terms. Alas, this prevents many forms that naturally arise, as say Fourier series expansions with multiple sinusoidal factors, which will become penalized heavily. However, when the terms are harmonically related, the penalty should be reduced as these likely point to structural order in a behavior. Also, for example, in fluid dynamics, the wave frequencies could be related as triads or quartets, which are also natural reinforcements so should come with a penalty reduction.

response:
https://chatgpt.com/share/6822ca5d-8854-8010-83cd-c163d435c6ea

In my own home-brewed model fitting of fluid dynamics, I rely on harmonics, standing-wave restrictions, and triads & quartets to find Fourier series groupings that symbolic regression algorithms give up on because of the high Pareto complexity.   If penalty relief is granted,  I guess this would have to be implemented as a custom loss function, if it has been done at all.

The presentation by Constantin is somewhat related to this, as one of his take-away messages is that finding such groupings as a "meta-law" may constrain the search space.  https://youtu.be/8o6jU-iBXbw?t=17518

**Comment by gkronber** [comment]
I apologize for reviving this old thread. 
The individual videos of the talks are online now.
Here is Miles' talk on PySR: https://youtu.be/g8HJ__rsFgE?feature=shared

Full playlist:  https://youtube.com/playlist?list=PLg7f-TkW11iVROREDn1NZPrWlvhRyDfMx&feature=shared
